CombinationService

Turntwo opened the pull request:
A service that allows for combining other services.  Options to handle suppression of duplicate event processing.  
This allows a setup where InMemoryService can be used for quick updates in the current process, while still sending the messages to other processes over the wcf interface using NetworkService.

Turntwo committed:
Merge pull request #2 from loraderon/master

Merge base Nvents changes

Turntwo committed:
Add CombinationService

Service that allows combining multiple services types (e.g. InMemory for
speed locally, and Network to also send events to connected
workstations).  Options to suppress rerunning actions for duplicate
events from multiple sources (None, UniqueOnly, or All).

Turntwo committed:
CombinationService - Fixes and refactor

Fix - DuplicateSuppression was overaggressive and filtered first
instance of an event because was incrementing the duplicate id prior to
checking if the event was a type that should be handled, but still
checking for each registration.  So it failed when more than one
subscription.
Removed - IUniqueEvent interface and handling - changed to use Equals
override on event classes, so no special interface needed.

loraderon commented:
Thanks, but I can't get CombinationServiceNetworkTests.DoesNotFilterDuplicateEventWhenUseEqualsOnly to pass?
Also could you document the difference with xml comments on DuplicateSuppressionOption?

Turntwo commented:
I will add the comments and check on the test.  I know it passed the last I tested it, but I’ll check it again.

From: Anders Wåglund [mailto:notifications@github.com]
Sent: Wednesday, December 04, 2013 4:26 AM
To: loraderon/nvents
Cc: Jason Christian
Subject: Re: [nvents] CombinationService (#7)

Thanks, but I can't get CombinationServiceNetworkTests.DoesNotFilterDuplicateEventWhenUseEqualsOnly to pass?
Also could you document the difference with xml comments on DuplicateSuppressionOption?

—
Reply to this email directly or view it on GitHubhttps://github.com/loraderon/nvents/pull/7#issuecomment-29800551.

Turntwo committed:
Combination Service - Fix DoesNotFilterDuplicateEventWhenUseEqualsOnly

Also document DuplicateSuppressionOptions in the enum itself.
Error was in the test method, not the CombinationService - the event was
being raised twice on each NetworkService, so changed to >=
expectedCount.

Turntwo committed:
Update comment about Duplicate Suppression

Turntwo commented:
Documented the DuplicateSuppressionOption, and fixed the test.

Thanks,

Jason

From: Anders Wåglund [mailto:notifications@github.com]
Sent: Wednesday, December 04, 2013 4:26 AM
To: loraderon/nvents
Cc: Jason Christian
Subject: Re: [nvents] CombinationService (#7)

Thanks, but I can't get CombinationServiceNetworkTests.DoesNotFilterDuplicateEventWhenUseEqualsOnly to pass?
Also could you document the difference with xml comments on DuplicateSuppressionOption?

—
Reply to this email directly or view it on GitHubhttps://github.com/loraderon/nvents/pull/7#issuecomment-29800551.
